streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamTransformValues.java

```diff
@@ -46,8 +58,104 @@ public KStreamTransformValuesProcessor(ValueTransformer<V, R> valueTransformer)
         }
 
         @Override
-        public void init(ProcessorContext context) {
-            valueTransformer.init(context);
+        public void init(final ProcessorContext context) {
+            valueTransformer.init(
+                new ProcessorContext() {
```

Comment:
Not sure either-- was thinking about inner class, too. Decided to just use anonymous as there will be no reuse and the class is quite simple, too.